internal/yggdrasil/yggdrasil.go

```diff
@@ -730,3 +736,41 @@ func (h *Handler) getDeviceSyslogLogConfig(ctx context.Context, edgeDevice *v1al
 		Protocol: proto,
 	}, nil
 }
+
+func GetDefaultMetricsReceiver() *models.MetricsReceiverConfiguration {
+	return &models.MetricsReceiverConfiguration{
+		RequestNumSamples: 30000,
+		TimeoutSeconds:    10,
+	}
+}
+
+func (h *Handler) getMetricsReceiverConfiguration(device *v1alpha1.EdgeDevice) (*models.MetricsReceiverConfiguration, error) {
+	result := GetDefaultMetricsReceiver()
+
+	if device != nil {
+		receiverConfig := device.Spec.MetricsReceiver
+		if receiverConfig != nil {
+			if receiverConfig.RequestNumSamples > 0 {
+				result.TimeoutSeconds = receiverConfig.TimeoutSeconds
+			}
+			if receiverConfig.TimeoutSeconds > 0 {
+				result.RequestNumSamples = receiverConfig.RequestNumSamples
+			}
+			if receiverConfig.URL != "" {
+				urlObj, err := url.Parse(receiverConfig.URL)
+				if err != nil {
+					return nil, err
+				}
+				if urlObj.Scheme != "http" {
+					return nil, fmt.Errorf("invalid url scheme for metrics receiver. got: %s. expected: http", urlObj.Scheme)
+				}
+				if urlObj.Host == "" {
+					return nil, fmt.Errorf("host cannot be empty for metrics receiver")
```

Comment:
Please  validate that in the webhook.

Reply:
if it is not merged yet then how can I use it? what do u suggest I do?

Reply:
https://issues.redhat.com/browse/ECOPROJECT-687. OK?

Reply:
:+1: